### PR TITLE
test(crux): domain agent and deployment tests

### PR DESCRIPTION
**Pull request by robot9706**
Added tests for agent server and deployment mapper.

### CODE REVIEW

File: web/crux/src/domain/agent.spec.ts [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+const AGENT_PUBLIC_KEY = 'public-key'
+
+const GrpcNodeConnectionMock = jest.fn().mockImplementation(() => ({
+  nodeId: AGENT_ID,
+  address: AGENT_ADDRESS,
+  connectedAt: new Date(),
+}))
+
+const observableComplete = <T>(it: Observable<T>) =>
+  new Promise(resolve => {
```
Comment (m8vago) [inline web/crux/src/domain/agent.spec.ts:27]: `const subscription = stream.subscribe(() => {})`
`expect(subscription.closed).toBe(true)`

File: web/crux/src/domain/agent.spec.ts [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+        address: AGENT_ADDRESS,
+        status: 'connected',
+        version: AGENT_VERSION,
+        connectedAt: agentConnection.connectedAt,
+        updating: false,
+      })
+    })
+
+    it('onDisconnected should send unreachable status', async () => {
+      const eventPromise = firstValueFrom(eventChannel.pipe(skip(1)))
```
Comment (m8vago) [inline web/crux/src/domain/agent.spec.ts:85]:
```
const resultObs = result.pipe(toArray())
const actual = await firstValueFrom(resultObs)
```

File: web/crux/src/domain/agent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+        version: AGENT_VERSION,
+        publicKey: AGENT_PUBLIC_KEY,
+      },
+      eventChannel,
+    )
+  })
+
+  describe('connection status', () => {
+    it('onConnected should send connection status', async () => {
+      const eventPromise = firstValueFrom(eventChannel)
```
Comment (m8vago) [inline web/crux/src/domain/agent.spec.ts:70]: you can simply await here

File: web/crux/src/domain/agent.spec.ts [hunk 4]
```diff
@@ -0,0 +1,380 @@
...
+
+      await expect(commandChannel).resolves.toEqual({
+        listSecrets: {
+          prefix: 'prefix',
+          name: 'name',
+        },
+      })
+
+      await expect(secrets).rejects.toThrow()
+    }, 10000)
```
Comment (m8vago) [inline web/crux/src/domain/agent.spec.ts:300]: please put a comment here why 10000
export the agent's constant as we discussed in person

File: web/crux/src/domain/deployment.spec.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  it('test containerNameFromImageName', () => {
+    expect(containerNameFromImageName('container')).toEqual('container')
+    expect(containerNameFromImageName('/container')).toEqual('container')
+    expect(containerNameFromImageName('/other/container')).toEqual('container')
+
+    expect(containerNameFromImageName('container:tag')).toEqual('container:tag')
+    expect(containerNameFromImageName('/container:tag')).toEqual('container:tag')
+    expect(containerNameFromImageName('/other/container:tag')).toEqual('container:tag')
+
+    // TODO (@robot9706): Real case?
```
Comment (m8vago) [inline web/crux/src/domain/deployment.spec.ts:56]: nope.

File: web/crux/src/domain/deployment.spec.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    expect(containerNameFromImageName('/')).toEqual('/')
+  })
+
+  it('test checkDeploymentCopiability should be true when status is not inProgress or preparing and the version is not rolling', () => {
+    DEPLOYMENT_STATUSES.forEach(deploymentStatus => {
```
Comment (m8vago) [inline web/crux/src/domain/deployment.spec.ts:61]: check if we can make test cases from an array of values

**Review by m8vago — CHANGES_REQUESTED** [review]

File: web/crux/src/domain/agent.spec.ts (hunk 2 above)
Comment (robot9706) [reply web/crux/src/domain/agent.spec.ts:85]: `toArray` requires the subject to complete which in this case it never does. Manually completing it works but it requires more code to do the same thing.

```
it('onConnected should send connection status', async () => {
      const eventPromise = firstValueFrom(eventChannel.pipe(toArray()))

      agent.onConnected()

      eventChannel.complete()
      await expect(eventPromise).resolves.toEqual({
        id: AGENT_ID,
        address: AGENT_ADDRESS,
        status: 'connected',
        version: AGENT_VERSION,
        connectedAt: agentConnection.connectedAt,
        updating: false,
      })
    })
```

**Review by m8vago — APPROVED** [review]

**Pull request merged**